src/main/java/net/countercraft/movecraft/combat/features/FlammableOverride.java

```diff
@@ -0,0 +1,186 @@
+package net.countercraft.movecraft.combat.features;
+
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import net.countercraft.movecraft.combat.MovecraftCombat;
+import net.countercraft.movecraft.util.Pair;
+import net.countercraft.movecraft.util.Tags;
+import org.apache.commons.lang.NotImplementedException;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+
+public class FlammableOverride {
+
+    public static Map<Material, Pair<Integer, Integer>> flammabilityOverride = null;
+    private static net.countercraft.movecraft.combat.features.FlammableOverride.FlammabilityNMS nmsInterface;
+
+    public static void load(@NotNull FileConfiguration config) {
+        try {
+            if (!config.contains("FlammabilityOverride"))
+                return;
+            var section = config.getConfigurationSection("FlammabilityOverride");
+            if (section == null)
+                return;
+
+            flammabilityOverride = new HashMap<>();
+            for (var entry : section.getValues(false).entrySet()) {
+                EnumSet<Material> materials = Tags.parseMaterials(entry.getKey());
+                for (Material m : materials) {
+                    int flammability;
+                    int encouragement;
+
+                    String valStr = entry.getValue().toString();
+                    try {
+                        String[] split = valStr.split(",");
+                        flammability = Integer.parseInt(split[0]);
+                        encouragement = flammability;
+                        if (split.length > 1) {
+                            encouragement = Integer.parseInt(split[1]);
+                        }
+                    } catch (NumberFormatException | NullPointerException ex) {
+                        MovecraftCombat.getInstance().getLogger()
+                                .warning("Unable to load " + m.name() + ": " + entry.getValue());
+                        continue;
+                    }
+                    flammabilityOverride.put(m, new Pair<>(flammability, encouragement));
+                }
+            }
+
+            String[] parts = Bukkit.getServer().getMinecraftVersion().split("\\.");
+            if (parts.length < 2)
+                throw new IllegalArgumentException();
+            int major_version = Integer.parseInt(parts[1]);
+            if (major_version < 20) {
+                nmsInterface = new FlammabilityNMS_SpigotMappings(); // Tested on 1.18.2 and 1.19.4
+            } else {
+                nmsInterface = new FlammabilityNMS_MojangMappings(); // Tested on 1.20.6
```

Comment:
If it's not been tested on these versions, the comments should be updated.

Reply:
Oh ye, uhm i forgot, i will do some testing on the weekend and will update the comments respectively